content/blog/recipes-1-1-0/index.Rmd

```diff
@@ -0,0 +1,239 @@
+
+In the below example, we specify a recipe where `x2` starts by being a character vector, but the recipe is prepped where `x2` is a numeric vector. This didn't produce any problems before,
+
+``` r
+data_template <- tibble(outcome = rnorm(10), x1 = rnorm(10), x2 = sample(letters, 10, T))
+
+rec <- recipe(outcome ~ ., data_template) %>%
+  step_bin2factor(all_numeric_predictors())
+
+data_training <- tibble(outcome = rnorm(1000), x1 = rnorm(1000), x2 = rnorm(1000))
+
+prep(rec, training = data_training)
+#> 
+#> ── Recipe ──────────────────────────────────────────────────────────────────────
+#> 
+#> ── Inputs
+#> Number of variables by role
+#> outcome:   1
+#> predictor: 2
+#> 
+#> ── Training information
+#> Training data contained 1000 data points and no incomplete rows.
+#> 
+#> ── Operations
+#> • Dummy variable to factor conversion for: x1 | Trained
+```
+
+but now we get an error detailing how the data is different:
+
+```{r}
+#| error: true
+data_template <- tibble(outcome = rnorm(10), x1 = rnorm(10), x2 = sample(letters, 10, T))
+
+rec <- recipe(outcome ~ ., data_template) %>%
+  step_bin2factor(all_numeric_predictors())
+
+data_training <- tibble(outcome = rnorm(1000), x1 = rnorm(1000), x2 = rnorm(1000))
+
+prep(rec, training = data_training)
+```
+
+In addition, we are exporting the two helper functions `recipes_ptype()` and `recipes_ptype_validate()` to extract and validate ptype information for a given recipe.
+
+```{r}
+recipes_ptype(rec)
+```
+
+Note that recipes created before version 1.1.0 don't contain any ptype information, and will not undergo checking. Rerunning the code to specify the recipe will add ptype information to the recipe.
+
+## Input checking in recipe()
+
+Every recipe you create start with a call to `recipe()`. We have relaxed the requirements of data frames, while increasing the feedback when something goes wrong.
+
+The data was previously passed through `model.frame()` inside the recipe, which restricted what could be handled. Previously prohibited input included data frames with list-columns or [sf](https://r-spatial.github.io/sf/) data frames. Both of these are now supported, as long as they are a `data.frame` object.
+
+```{r}
+data_listcolumn <- tibble(
+  y = 1:4,
+  x = list(1:3, 4:6, 3:1, 1:10)
+)
+
+recipe(y ~ ., data = data_listcolumn)
+```
+
+```{r}
+library(sf)
+pathshp <- system.file("shape/nc.shp", package = "sf")
+data_sf <- st_read(pathshp, quiet = TRUE)
+
+recipe(AREA ~ ., data = data_sf)
+```
+
+We are excited to see what people can do with these new options.
+
+Another way to specify a recipe is to use `add_role()` and `update_role()`. But if you are not careful, you can end up in situations where the same variable is labeled as both the outcome and predictor.
```

Comment:
```suggestion
Another way to specify a recipe is to use `add_role()` and `update_role()`. But if you were not careful, you could end up in situations where the same variable is labeled as both the outcome and predictor.
```
Could you smooth the transition here a bit? Something about how the call to `recipe()` specifies the roles but you can change these later?